plugins/renepay/pay.c

```diff
@@ -0,0 +1,1798 @@
+				   pay_plugin->chan_extra_map);
+#if DEVELOPER
+	plugin_set_memleak_handler(p, memleak_mark);
+#endif
+
+	background_settimer();
+	return NULL;
+}
+
+
+// /* TODO(eduardo): an example of an RPC call that is not bound to any command. */
+//static
+//struct command_result* getinfo_done(struct command *cmd UNUSED,
+//			 const char *buf,
+//			 const jsmntok_t *result,
+//			 void* pp  UNUSED)
+//{
+//	struct node_id id;
+//	const jsmntok_t *id_tok = json_get_member(buf,result,"id");
+//	json_to_node_id(buf,id_tok,&id);
+//
+//	plugin_log(pay_plugin->plugin,LOG_DBG,
+//		"calling %s, nodeid = %s",
+//		__PRETTY_FUNCTION__,
+//		type_to_string(tmpctx,struct node_id,&id));
+//
+//	return command_still_pending(NULL);
+//}
+
+static void background_settimer(void)
+{
+	pay_plugin->rexmit_timer
+		= tal_free(pay_plugin->rexmit_timer);
+	pay_plugin->rexmit_timer
+		= plugin_timer(
+			pay_plugin->plugin,
+			time_from_msec(2000),
+			background_timer_kick, NULL);
+}
+
+static void background_timer_kick(void * p UNUSED)
+{
+	// plugin_log(pay_plugin->plugin,LOG_DBG,"calling %s",__PRETTY_FUNCTION__);
+	background_settimer();
+
+	// /* TODO(eduardo): an example of an RPC call that is not bound to any command. */
+	// struct out_req * req = jsonrpc_request_start(pay_plugin->plugin,
+	// 				NULL,
+	// 				"getinfo",
+	// 			    	getinfo_done,
+	// 			    	getinfo_done,
+	// 			    	NULL);
+	// send_outreq(pay_plugin->plugin, req);
+}
+
+static void renepay_settimer(struct renepay * renepay)
+{
+	renepay->rexmit_timer = tal_free(renepay->rexmit_timer);
+	renepay->rexmit_timer = plugin_timer(
+			pay_plugin->plugin,
+			time_from_msec(TIMER_COLLECT_FAILURES_MSEC),
+			timer_kick, renepay);
+}
+
+/* Happens when timer goes off, but also works to arm timer if nothing to do */
+static void timer_kick(struct renepay * renepay)
+{
+	struct payment * const p = renepay->payment;
+	plugin_log(pay_plugin->plugin,LOG_DBG,"calling %s",__PRETTY_FUNCTION__);
+
+	switch(p->status)
+	{
+		/* Some flows succeeded, we finish the payment. */
+		case PAYMENT_SUCCESS:
+			renepay_success(renepay);
+		break;
+
+		/* Some flows failed, we retry. */
+		case PAYMENT_FAIL:
+			payment_assert_delivering_incomplete(p);
+			try_paying(renepay->cmd,renepay,false);
+		break;
+
+		/* Nothing has returned yet, we have to wait. */
+		case PAYMENT_PENDING:
+			payment_assert_delivering_all(p);
+			renepay_settimer(renepay);
+		break;
+	}
+}
+
+/* Sometimes we don't know exactly who to blame... */
+static struct command_result *handle_unhandleable_error(struct renepay * renepay,
+							struct pay_flow *flow,
+							const char *what)
+{
+	struct payment * const p = renepay->payment;
+	plugin_log(pay_plugin->plugin,LOG_DBG,"calling %s",__PRETTY_FUNCTION__);
+	size_t n = tal_count(flow);
+
+	/* We got a mangled reply.  We don't know who to penalize! */
+	debug_paynote(p, "%s on route %s", what, flow_path_to_str(tmpctx, flow));
+
+	// TODO(eduardo): does LOG_BROKEN finish the plugin execution?
```

Comment:
No, plugin_err does.  But it would not be good to do that here, just give up maybe?

Reply:
Got it!